python/cudf/cudf/core/column/numerical.py

```diff
@@ -205,6 +205,19 @@ def as_timedelta_column(
             ),
         )
 
+    def as_decimal_column(
+        self, dtype: Dtype, **kwargs
+    ) -> "cudf.core.column.DecimalColumn":
+        if is_integer_dtype(self.dtype):
+            raise NotImplementedError(
+                "Casting from integer types to decimal "
+                "types not currently supported"
+            )
+        result = libcudf.unary.cast(self, dtype)
+        if isinstance(dtype, cudf.core.dtypes.Decimal64Dtype):
+            result.dtype.precision = dtype.precision
```

Comment:
Why is it required to update precision in result ?

Reply:
The libcudf unary cast always returns a precision of 18 so it needs to be patched with the correct precision on the python side